Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Culturally weighted CoA colours (WIP)

## Changelog:
Added colour weights to human culture CoAs. A higher weight number corresponds to a higher probability of selection.
Below are the custom weights. These do not include include the base templates for normal and metal colours.

<details>
<summary> Click to show details </summary>

- **Stormwind**
  - Main Colours
    - Cyan Silverhand
    - Blue
  - Emblem Colours
    - White
    - Yellow
- **Lordaeron**
  - Main Colours
    - Purple
    - Cyan Silverhand
    - Grey Dark
    - Blue
  - Emblem Colours
    - Yellow
	- White
- **Arathor**
  - Main Colours
    - Purple
    - Black
    - Brown Waycrest
    - Red
  - Emblem Colours
    - Yellow 
	- White
- **Alterac**
  - Main Colours
    - Purple
    - Black
    - Brown Dark
  - Emblem Colours
    - Yellow Alterac
    - Orange Light
- **Gilneas**
  - Main Colours
    - Brown Dark
    - Blue Gilneas
    - Black Gilneas
  - Emblem Colours
    - Orange Light
    - Yellow
    - Red Gilneas
- **Kul Tiras**
  - Main Colours
    - Blue Gilneas
    - Purple Tidesage
    - Brown Waycrest
    - Aqua Stormsong
    - Orange Ashvane
    - Green Kultiras
  - Emblem Colours
    - Yellow
    - Silver
    - Yellow Kultiras
- **Theramore**
  - Main Colours
    - Brown Waycrest
    - Aqua Stormsong
    - Orange Ashvane
    - Purple
    - Green Kultiras
    - Blue
  - Emblem Colours
    - Silver
    - Yellow Kultiras
    - White
- **Dalaran/Kirin Tor**
  - Main Colours
    - Blue Light
    - Purple
  - Emblem Colours
    - Silver
    - White
- **Hillsbrad**
  - Main Colours
    - Purple
    - Red
    - Brown
    - Green
    - Blue Light	
  - Emblem Colours
    - White
- **Wastewander**
  - Main Colours
    - Black
    - Orange Light
    - Red Light
  - Emblem Colours
    - Yellow Wastewander

</details>


## Developer changelog:
**Testing human changes; will extend to other culture groups once reviewed.**

Prevents randomised CoAs from having similar colours to special selections due to default `normal` and `metal` lists.
Added cultural specific colours to wc_colors.txt.
Triggers linked to graphical_culture_gfx in common/culture. 
Added coa graphical cultures for dalaran, hillsbrad, and theramore.
  - Added Dalaran, Hillsbrad and Thereamore to 'charges' and 'hr_charges'
  - Added Gilneas to 'pagan_single_charge_list'

- Dalaran emblem graphics: 
"ce_kirintor_eye.dds"

- Hillsbrad emblem graphics: 
"ce_lordaeron.dds"
"wc_stormwind_lion.dds"
"ce_gilneas_moon.dds"
"ce_stromgarde_fist.dds"
"ce_alteran_eagle.dds"
"ce_kul_tiras.dds"
"ce_kirintor_eye.dds"

- Theramore emblem graphics: 
"ce_kul_tiras.dds"
"ce_kirintor_eye.dds"

### In Progress
- [x] Triggers do not apply to historical dynasties. **IMPORTANT**


## Tests:


# How to test:
Look for generated minor nobles and/or run the game long enough that ahistorical human CoAs begin cropping up. Check for increased frequency of colours as specified below. Also suggest more colours/hues/weights. If you wish to see the colours outside of the game, you can key them into a RGB webite using the codes in `wc_colors.txt` and `default_colors.txt`.

<details>
<summary> Click to show image </summary>

![image](https://user-images.githubusercontent.com/77478662/149676381-3ec82286-3ccf-4871-b4c3-77b937396051.png)

</details>
